code/modules/power/singularity/emitter.dm

```diff
@@ -144,31 +163,28 @@
 
 /obj/machinery/power/emitter/interact(mob/user)
 	add_fingerprint(user)
-	if(welded)
-		if(!powernet)
-			to_chat(user, "<span class='warning'>\The [src] isn't connected to a wire!</span>")
-			return TRUE
-		if(!locked && allow_switch_interact)
-			if(active == TRUE)
-				active = FALSE
-				to_chat(user, "<span class='notice'>You turn off [src].</span>")
-			else
-				active = TRUE
-				to_chat(user, "<span class='notice'>You turn on [src].</span>")
-				shot_number = 0
-				fire_delay = maximum_fire_delay
-
-			message_admins("Emitter turned [active ? "ON" : "OFF"] by [ADMIN_LOOKUPFLW(user)] in [ADMIN_VERBOSEJMP(src)]")
-			log_game("Emitter turned [active ? "ON" : "OFF"] by [key_name(user)] in [AREACOORD(src)]")
-			investigate_log("turned [active ? "<font color='green'>ON</font>" : "<font color='red'>OFF</font>"] by [key_name(user)] at [AREACOORD(src)]", INVESTIGATE_SINGULO)
-
-			update_appearance()
-
-		else
-			to_chat(user, "<span class='warning'>The controls are locked!</span>")
-	else
+	if(!welded)
 		to_chat(user, "<span class='warning'>[src] needs to be firmly secured to the floor first!</span>")
 		return TRUE
+	if(!powernet)
+		to_chat(user, "<span class='warning'>\The [src] isn't connected to a wire!</span>")
+		return TRUE
```

Comment:
why return true? shouldn't this be false to maintain behavior with the old code?

Reply:
yes i'll change them to be false